### DISCUSSION TITLE
Is helm-controller capable to detect drift?

### DISCUSSION THREAD

**Discussion by wibed** [fluxcd/flux2#2873] [Q&A] [answered]
the following assumes fluxcd does detect and correct drift: 
https://github.com/fluxcd/kustomize-controller/issues/352

in contrast to:
https://github.com/fluxcd/flux2/discussions/2812

is only helm-controller affected by the lack of detecting drift,
or are terms used interchangeably for different kinds of drift?

**Comment by stefanprodan** [comment, marked as answer]
Flux helm-controller drift detection of resources contained in a chart is in the works. I expect this feature to be ready by end of summer.

Update: Helm drift detection and correction is now available in Flux v2.0.0-rc.1 https://fluxcd.io/flux/cheatsheets/bootstrap/#enable-helm-drift-detection

**Comment by kingdonb** [comment]
The discussion about drift detection in Helm Controller is ongoing here:

https://github.com/fluxcd/helm-controller/issues/186

Outside of Helm Controller, Flux does not have any problem with detecting and correcting drift. In here it's problematic because of several Helm-specific issues like: postrender kustomizations, helm lifecycle hooks (jobs), that are difficult to capture and account for outside of Helm's imperative release process.

As it's been said, the work is in progress and should be coming soon.

**Comment by steeling** [comment]
The docs here very clearly state that one of the benefits of flux with helm is that it provides drift detection. This is very misleading if this is not the case

https://fluxcd.io/flux/use-cases/helm/

**Reply by wibed** [reply]
> The docs here very clearly state that one of the benefits of flux with helm is that it provides drift detection. This is very misleading if this is not the case
> 
> https://fluxcd.io/flux/use-cases/helm/

please due diligence and read the docs

**Comment by steeling** [comment]
Hey @wibed, I’ve tried reading the docs but may have missed something

> Automated drift detection between the desired and actual state of your operations

is mentioned as a benefit in the link I included above, but this issue shows it as experimental. https://github.com/fluxcd/helm-controller/issues/643, and sounds like a drift of one object will apply an entire upgrade on the helm release which may have undesired side effects. 

If I’m missing something would you be able to point me in the right direction?

**Comment by wibed** [comment]
i have not yet implemented it.

as you stated, drift detection is experimental.
please open an issue if you feel like the software is misbehaving in some way.
it may help the development process
